Props Convert

ssorallen opened the pull request:
- Use updated `Polymer.dom` for distributed nodes
- Convert to `properties` object

Commit:
Use updated `Polymer.dom` for distributed nodes

`this.localDom` no longer has an API for distributed nodes.

Commit:
Convert to `properties` object

`computedProperties` and `configure` merge into `properties`.

Commit:
Prefer `computed` for computed func names per guide

cdata commented on core-header-panel.html:
```diff
@@ -139,12 +139,24 @@
+      properties: {
+
+        animateDuration: {
+          value: 200
```
No `type`?

cdata commented on core-header-panel.html:
```diff
@@ -139,12 +139,24 @@
+        },
+
+        atTop: {
+          value: true
```
Default value of `true` is not supported for booleans, strictly speaking. Is this an effectively private property?

ssorallen replied:
Yup. By omitting `type` I'm sort of making these "private". I could just assign them in `ready` too?

cdata commented on core-header-panel.html:
```diff
@@ -288,7 +300,7 @@
 
       get header() {
-        return this.localDom.distributedNodes(this.$.headerContent)[0];
+        return Polymer.dom.distributedNodes(this.$.headerContent)[0];
```
This API will be different by EOD.

Commit:
Use updated `Polymer.dom` API

The breaking change occurred in commit
https://github.com/Polymer/polymer/commit/28c6f91b0d0bb314aa8f85954e2ca9de459cd0a3
